### ISSUE TITLE
syntax/block highlighted issue

### ISSUE THREAD

**Issue by tcloaa** [MacDownApp/macdown#682]
Macdown is pretty awesome, but I found the code block parsing is not as convenient as other editors do (github wiki page, typora.)

1. Everytime you want to write a code block, you have to make sure that there is a newline right above it.
![image](https://cloud.githubusercontent.com/assets/8750937/19741535/a67386fa-9b86-11e6-847b-9728b022932d.png)

Otherwise the parsing won't work.
![image](https://cloud.githubusercontent.com/assets/8750937/19741564/c45bdd98-9b86-11e6-9344-300536724362.png)

which I think that this setting is not aligned with how github/typora do. So in order to use Macdown, I have to insert a newline right above every code block I write...

2. Github and typora accept "C" (capital letter) as parsing c block, while macdown doesn't.. which is also not aligned with other editors as well..

Just a few comment! And thanks for building up such nice tool.

**Comment by franklinyu** [comment]
Yes. According to [CommonMark](http://spec.commonmark.org/0.26/#fenced-code-blocks) (emphasize mine):

> A fenced code block may interrupt a paragraph, and does *not* require a blank line either before or after.

**Comment by uranusjr** [comment]
For issue 1 I don’t think that is a bug, but a disagreement in specs. But the situation is as mentioned.

For issue 2, MacDown actually has support for custom language names; it’s just not exposed in the GUI. Open `MacDown.app/Contents/Resources/syntax_highlighting.json` and you’ll see a list of available aliases I added besides Prism’s built-in language names. Just add what you want to it. (And remember to keep a backup because a MacDown update will override this file.)

**Comment by Hipska** [comment]
Is there any improvement yet for Issue 1? I think I found a related issue on their project page: https://github.com/hoedown/hoedown/issues/191

**Comment by cucumbur** [comment]
Very annoying "feature", Hoedown doesn't seem amenable to commonmark. I love MacDown but feature and syntax disparity between markdown processors and clients are getting ridiculous.
